madcad/draft.py

```diff
@@ -0,0 +1,194 @@
+from typing import Tuple
+
+import numpy as np
+
+from madcad.prelude import *
+import madcad as cad
+from madcad.mesh.container import edgekey
+
+
+def draft_angles(mesh: Mesh, draft_direction: vec3, **kwargs):
+	"""
+	returns:
+	a list of draft angles for each face when compared to the draft direction
+	"""
+	return [
+		angle_vectors(draft_direction, face_n, **kwargs)
+		for face_n in mesh.facenormals()
+	]
+
+
+def angle_vectors(v1: vec3, v2: vec3, degrees=False) -> float:
+	"""
+	returns angle between two vectors
+	"""
+	n1 = normalize(v1)
+	n2 = normalize(v2)
+	angle = np.arccos(dot(n1, n2))
+	if not degrees:
+		return angle
+	return np.rad2deg(angle)
+
+
+def offset_vector(n1: vec3, n2: vec3):
+	"""
+	if edge1 and edge2 is offset a distance(t) along their normals
+	returns the intersection offset/t
+	"""
+	n3 = normalize(n1 + n2)
+	cos = dot(n1, n3)
+	scale = 1 / cos
+	return n3 * scale
+
+
+def edges_with_points(edges, pids, only=True):
+	"""
+	args:
+		edges: list[vec2]
+		pids: list[int], white list of point indices
+		only: bool, toggles if edges must have *only* white listed points
+
+	returns:
+		a subset of edges that have specifed points
+	"""
+	if only:
+		bool_reduce = all
+	else:
+		bool_reduce = any
+
+	edges = tlist(
+		[e for e in edges if bool_reduce((e[0] in pids, e[1] in pids))],
+		uvec2,
+	)
+
+	return edges
+
+
+def draft(mesh: Mesh, axis: Axis, angle: float) -> Mesh:
+	"""
```

Comment:
The end-user functions in this module (`draft()` and `draft_side()`), as well as the module itself, are lacking a precise docstring and eventually screenshots for the sphinx doc.
Of course that's something really meaningful to make only when the development is finished

For the screenshots of course there will be the theming issue: to be consistent with the rest of the docs I shall have the same color pack as mine ;)